packages/libs/client-utils/src/core/estimateGasPrice/utils.ts

```diff
@@ -0,0 +1,155 @@
+    avg: number; // average gas price in the block transactions (does not consider usage)
+    median: number; // median gas price in the block transactions
+  };
+}
+
+/**
+ * Calculates gas information for a given set of blocks.
+ * @param blocks - An object containing block information.
+ * @returns An array of gas information for each block.
+ */
+export function calculateGasInformation(
+  blocks: IBlockChainData,
+  { maxBlockCapacity = MAX_BLOCK_CAPACITY } = {},
+) {
+  const gasData = blocks.items.map((block) => {
+    const txData = block.payloadWithOutputs.transactions.map(
+      ([payload, result]) => {
+        const tx = JSON.parse(base64UrlDecode(payload));
+        const command = JSON.parse(tx.cmd);
+        const gasConsumed = JSON.parse(base64UrlDecode(result)).gas;
+        return {
+          hash: tx.hash,
+          gasPrice: command.meta.gasPrice,
+          gasLimit: command.meta.gasLimit,
+          consumedGas: gasConsumed,
+          gasPaid: command.meta.gasPrice * gasConsumed,
+        };
+      },
+    );
+
+    const totalGasConsumed = txData.reduce((acc, tx) => {
+      return acc + tx.consumedGas;
+    }, 0);
+
+    const totalGasLimit = txData.reduce((acc, tx) => {
+      return acc + tx.gasLimit;
+    }, 0);
+
+    const totalGasPaid = txData.reduce((acc, tx) => {
+      return acc + tx.gasPaid;
+    }, 0);
+
+    const txGasPrice = txData
+      .map((tx: { gasPrice: number }) => tx.gasPrice)
+      .sort((a, b) => a - b);
+    const minGasPrice = txGasPrice[0] ?? 0;
+    const maxGasPrice = txGasPrice[txGasPrice.length - 1] ?? 0;
+
+    const gasInformation: IBlockGasInformation = {
+      blockHeight: block.header.height,
+      totalGasConsumed,
+      totalGasLimit,
+      totalGasPaid,
+      txCount: txData.length,
+      blockGasUsedPercent: (totalGasConsumed * 100) / maxBlockCapacity,
+      gasPriceStats: {
+        min: minGasPrice,
+        max: maxGasPrice,
+        avg: average(txGasPrice),
+        median: median(txGasPrice),
+      },
+    };
+    return gasInformation;
+  });
+
+  const sorted = gasData.sort((a, b) => b.blockHeight - a.blockHeight);
+
+  return sorted;
+}
+
+/**
+ * Calculates the gas price based on the provided gas data.
+ * @param gasData - An array of gas data for each block.
+ * @param considerCongestion - A boolean indicating whether to consider usage in the calculation.
+ * @returns The estimated gas price.
+ * @example
+ * const estimatedGasPrice = calculateGasPrice(gasData);
+ * console.log('Estimated gas price:', estimatedGasPrice);
+ * // Output: Estimated gas price: 0.00000001
+ */
+export function calculateGasPrice(
+  gasData: Array<{
+    txCount: number;
+    gasPriceStats: { min: number };
+  }>,
+) {
+  const mins = gasData
+    .filter((b) => b.txCount > 0)
+    .map((data) => data.gasPriceStats.min)
+    .sort((a, b) => a - b);
+
+  const estimate = median(mins);
+
+  return estimate || MINIMUM_GAS_PRICE;
```

Comment:
If it's a minimum instead of default, shouldn't this be `Math.max(estimate || 0, MINIMUM_GAS_PRICE)`